### PR TITLE
add hospitalizations to reichlab formatter, changes from #125, cleanup

### PR DESCRIPTION
**Pull request by dajmcdon**
- This should format the hospitalization forecasts correctly as well. 
- Many R checks fail, from previous updates. Most associated with corrections.
- Check that corrections are working in Hosp Production

### CODE REVIEW

**Comment by dshemetov** [comment]
Didn't follow every line of code exactly, but switching over to this branch stitched the results together correctly in production.

EDIT: Read it over in more detail and it looks good.

**Review by dshemetov — DISMISSED** [review]

File: utilities/zookeeper/R/reichlab-formatter.R
Comment (brookslogan) [inline utilities/zookeeper/R/reichlab-formatter.R:81]: Quantile forecasts are filtered to the accepted quantile levels for cases, but maybe not for deaths and hospitalizations.

Comment (brookslogan) [reply utilities/zookeeper/R/reichlab-formatter.R:81]: Additionally, around here, if there is already a point prediction provided, it might not be recognized and another point forecast based on the median might be tacked on.

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [inline utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: Might need to be 2020-01-30 for the 4-wk-ahead?

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [inline utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: Again, here.  2020-01-23 vs 2020-01-30 for 4 wk ahead

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [inline utilities/zookeeper/tests/testthat/test-reichlab-formatter.R:18]: pcard might not be used in this test

Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R:32]: --- stopped looking at this point.  will pick up from here later ---

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: or 2020-01-06 for 4 day ahead?

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: or 2020-01-06 for 4 day ahead?

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: It appears that it should be the latter.  This highlights that the formatter is not validating the `target_end_date`s.

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: Should be the latter.

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R:18]: --> removing or commenting out definition of pcard is fine

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: Additionally, the formatter is not validating that the `geo_value`s meet its expectations; e.g., it could get state fips codes and attempt to convert them "from abbrevations" to fips codes, resulting in NAs.  The existing partial guard against this is the check for the `"predictions_cards"` class, but, e.g., `as.predictions_cards` looks like it may still allow a user to encounter this case.

File: utilities/zookeeper/R/reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/R/reichlab-formatter.R:81]: Filed away in #126 and #128.  These are just validation issues, so it's fine to merge this without addressing them.  However, I believe #128 or a similar issue has caused issues in some forecasting scripts, where the baseline model outputting a point prediction when all the other forecasters did not actually broke some ensemble geo-reaggregation code.

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: Addressed.  Validation issue noted in #127.

File: utilities/zookeeper/tests/testthat/test-reichlab-formatter.R
Comment (brookslogan) [reply utilities/zookeeper/tests/testthat/test-reichlab-formatter.R]: Addressed.  Validation issue noted in #127.

**Comment by brookslogan** [comment]
This looks good to me.  Tests pass locally for me.  @dshemetov feel free to merge.  (Not sure if you can unassign yourself to review and re-approve, then merge?  Or if merging the PR #125 into this disqualified us from acting as reviewers.  Anyway, to consider this fully reviewed you might want to check that I didn't break the tests/build for you; my last commit here only touched the tests.)

**Comment by dshemetov** [comment]
While I'm not a codeowner, you have the power to give an approving review and merge. How we get it to re-request a review from you though, I'm not sure. Maybe if you go to the file compare, you'll see an opportunity for another review up in the top right?

**Review by brookslogan — APPROVED** [review]

**Comment by brookslogan** [comment]
I did a little additional cleanup in this PR (in addition to your cleanup inherited from #125) to satisfy some checks and improve some docs.

Re-request worked.  Not sure if I actually had to remove my own assignment first or not.  GitHub seems a little sassy or nervous with the "self-requested a review" but allows the merge.

**Comment by dshemetov** [comment]
Nice thanks! Let's delete `add-hosp` too, to keep the branches clean.

**Pull request merged**
